hubblestack/extmods/grains/custom_grains_pillar.py

```diff
@@ -0,0 +1,69 @@
+'''
+HubbleStack Custom Grains and Pillar
+
+Allows for fetching custom grain and pillar data from a local salt-minion via
+salt-call
+
+:maintainer: HubbleStack
+:platform: All
+:requires: SaltStack
+'''
+
+import re
+import salt.modules.cmdmod
+import logging
+
+log = logging.getLogger(__name__)
+
+__salt__ = {
+        'cmd.run': salt.modules.cmdmod._run_quiet,
+        'config.get': salt.modules.config.get,
+}
+
+
+def populate_custom_grains_and_pillar():
+    '''
+    Populate local salt-minion grains and pillar fields values as specified in
+    config file.
+
+    For example:
+
+        custom_grains_pillar:
+          grains:
+            - selinux: selinux:enabled
+            - release: osrelease
+          pillar:
+            - ntpserver: network_services:ntpserver
+
+    Note that the core grains are already included in hubble grains -- this
+    is only necessary for custom grains and pillar data.
+    '''
+    log.debug('Fetching custom grains and pillar details')
+    grains = {}
+    salt.modules.config.__opts__ = __opts__
+    custom_grains = __salt__['config.get']('custom_grains_pillar:grains', [])
+    for grain in custom_grains:
+        for key in grain:
+            if _valid_command(grain[key]):
+                value = __salt__['cmd.run']('salt-call grains.get {0}'.format(grain[key])).split('\n')[1].strip()
```

Comment:
this feels like a mistake could cause major problems... What about something like:

```
__salt__['cmd.run'](['salt-call', 'grains.get', grain[key]).split( ... )
```
?

Also, why cmd.run? can't we call grains.get directly? This is because it's coming from salt, not hubble, right?

Reply:
Salt, by default, uses shlex.split to split the command into an array to avoid command injection vulnerabilities. So we're safe unless you pass in `python_shell=True`.

And yes, this is for getting info from the local salt minion. Mainly pillar data, but also potentially custom grains.